### DISCUSSION TITLE
Help getting psalm to understand `filter`

### DISCUSSION THREAD

**Discussion by bendavies** [azjezz/psl#226] [Q&A] [answered]
Hi there!

Take a look at this very basic example.
the `filter` function here is as per `Psl\Vec\filter`.
https://psalm.dev/r/550a18c95b

What doesn't psalm like about this?

Thanks!

**Comment by azjezz** [comment, marked as answer]
This looks like a bug in Psalm rather than PSL, I have created an issue in Psalm and see how that goes 🤞 

https://github.com/vimeo/psalm/issues/6463

**Reply by bendavies** [reply]
thanks! i thought it might be. just thought i'd confirm here first.

**Reply by azjezz** [reply]
this issue is related to psalm, you can subscribe to vimeo/psalm#6463 for updated 👍
